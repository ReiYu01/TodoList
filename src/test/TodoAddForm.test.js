import React from 'react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import TodoAddForm from '../components/todo/TodoAddForm'

describe('Should Render TodoAddForm', () => {
  const mockStore = configureStore()
  const store = mockStore()

  it('renders correctly', () => {
    const component = renderer.create(<TodoAddForm store={store} />)
    let tree = component.toJSON()
    // tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    // })

    // it('onChange correctly', () => {
    tree.props.onKeyPress()
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  // let chart = tree.toJSON()
  // expect(chart).toMatchSnapshot()

  // chart.props.onKeyPress()
  // // re-rendering
  // chart = tree.toJSON()
  // expect(chart).toMatchSnapshot()
})
