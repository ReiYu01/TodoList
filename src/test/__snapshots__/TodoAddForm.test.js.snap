// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render TodoAddForm renders correctly 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="todoInput"
  >
    Please input
  </label>
  <input
    className="form-control"
    id="todoInput"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Please input"
    type="text"
    value=""
  />
</div>
`;
